src/bootstrap/tool.rs

```diff
@@ -250,6 +250,10 @@ pub fn prepare_tool_cargo(
         }
     }
 
+    // cargo doesn't set this env var yet, but it's needed by expect-test
+    // to build correct absolute paths for test_data files
+    cargo.env("CARGO_WORKSPACE_DIR", &dir);
```

Comment:
Since this is a non-standard thing, maybe we should only set that in `test::RustAnalyzer`? And even if it becomes standard, other crates in the tree might actually want the overall rust workspace root.

Reply:
That's a great point, that's an easy change to make — I'll do that for the final PR